silence/silence_test.go

```diff
@@ -337,12 +337,12 @@ func TestSilences_Maintenance_SupportsCustomCallback(t *testing.T) {
 	gosched()
 
 	// Before the first tick, no maintenance executed.
-	clock.Add(9 * time.Second)
+	clock.Advance(9 * time.Second)
 	require.EqualValues(t, 0, calls.Load())
 
 	// Tick once.
-	clock.Add(1 * time.Second)
-	require.EqualValues(t, 1, calls.Load())
+	clock.Advance(1 * time.Second)
+	require.Eventually(t, func() bool { return calls.Load() == 1 }, 5*time.Second, time.Second)
```

Comment:
This is now required as otherwise the `require.EqualValues` is run before the goroutine. This didn't happen with `benbjohnson/clock` as it included additional synchronization code to ensure all timers fired before `clock.Add` returned. This made tests easier to write, but is not how the `time` package behaves in real life.